lib/core/utils/query-selector-all-filter.js

```diff
@@ -97,6 +98,17 @@ function matchExpressions(domTree, expressions, filter) {
  */
 function querySelectorAllFilter(domTree, selector, filter) {
   domTree = Array.isArray(domTree) ? domTree : [domTree];
+
+  // see if the passed in node is the root node of the tree and can
+  // find nodes using the cache rather than looping through the
+  // the entire tree
+  const nodes = getNodesMatchingSelector(domTree, selector, filter);
```

Comment:
Passing an expression makes testing harder as instead of a simple string I need to parse the selector into the expression first then pass it to the function. I don't feel that passing an expression makes it any more consistent either.